src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,68 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        // Try to resolve implementation for the given interface
+        Class<?> implClass = findImplementation(interfaceClazz);
+
+        // Check if class is annotated with @Component
+        if (!implClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + implClass.getName()
+                    + " is not marked as @Component");
+        }
+
+        // Check if an instance already exists
+        if (instances.containsKey(implClass)) {
+            return instances.get(implClass);
+        }
+
+        // Create a new instance and inject dependencies
+        Object instance = createInstance(implClass);
+        instances.put(implClass, instance);
+        return instance;
+    }
+
+    private Object createInstance(Class<?> implClass) {
+        try {
+            Object instance = implClass.getDeclaredConstructor().newInstance();
+
+            // Inject dependencies into fields annotated with @Inject
+            for (Field field : implClass.getDeclaredFields()) {
+                if (field.isAnnotationPresent(Inject.class)) {
+                    field.setAccessible(true);
+                    Object fieldInstance = getInstance(field.getType());
+                    field.set(instance, fieldInstance);
+                }
+            }
+
+            return instance;
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create instance of " + implClass.getName(), e);
```

Comment:
It is better to replace many exceptions that have a common parent with a general parental exception to simplify the catch block. Consider using `ReflectiveOperationException` which is a common superclass of the exceptions thrown by reflective operations in the `java.lang.reflect` package.